### ISSUE TITLE
Allow to specify env webpack variables 

### ISSUE THREAD

**Issue by danielo515** [AnomalyInnovations/serverless-bundle#49]
Hello, and thank you for this plug in, it has served me well so far.
But now I am on the situation that some random node dependency is doing weird things and I need to setup an environment variable so webpack does not bundle a part of the code. I took a look at the docs and there is no such option. Would you mind adding an env section to the configuration section? 

Thanks!

**Comment by jayair** [comment]
Can you share more details on this? It would good if you can share a sample repo so I can look into what to do.

**Comment by danielo515** [comment]
Well, I don't think an example repository could help for requesting a new feature, but I can give you more details.

Basically what I want is to be able to specify values for this plugin:

https://webpack.js.org/plugins/environment-plugin/#usage-with-default-values

So I can set some env variables for the code that is being bundled.

**Comment by jayair** [comment]
I'm not completely clear about this. Can you not set these variables in your `serverless.yml` as environment variables?

**Comment by danielo515** [comment]
Yes, I can, but those variables are not for changing the behavior of webpack, it is for the bundled code.
There are parts of javascript code that has conditional code based on env variables, but those variables needs to be included during the build phase. For that, webpack needs to be configured to make those variables available to the bundled code.
Basically, all you need to do is inject the following plugin in the list of plugins:

```
new webpack.EnvironmentPlugin(['NODE_ENV', 'DEBUG']);
```

That will make NODE_ENV and DEBUG variables available on the bundled code, not only during webpack runtime.
Does that make it clearer?

**Comment by clementAC** [comment]
setting your serverless.yml like this will make env var available in the lambda runtime context:

``` yaml
provider:
  name: aws
  runtime: nodejs12.x
  memorySize: 256
  environment:
    NODE_ENV: ${opt:stage}
    DEBUG: ${self:custom.environmentVar.DATABASE.${opt:stage}.PGUSER}
   ...
functions:
... 
```

**Comment by danielo515** [comment]
@clementAC my target is not the lambda runtime, is web-pack build-time context, and there is no way you can do it without the change I spotted above on the web-pack configuration.
I'm starting to think that making a PR with my proposed changes will actually be easier than trying to explain it

**Comment by averyyip** [comment]
@danielo515 Curious if you ended up implementing this yourself? I'm running into the same exact problem!

**Comment by danielo515** [comment]
@averyyip nope, I just decided to "opt-out" and build a webpack pipeline myself, then upload that code. But this was 1 year ago, so the info it's not fresh on my mind.
